opentracing-shim/src/opentracingshim/__init__.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from contextlib import contextmanager
+
+import opentracing
+
+from opentelemetry.trace import Event
+from opentracingshim import util
+
+
+def create_tracer(tracer, scope_manager=None):
+    return TracerWrapper(tracer, scope_manager)
+
+
+class SpanContextWrapper(opentracing.SpanContext):
+    def __init__(self, otel_context):
+        self._otel_context = otel_context
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `SpanContext` object."""
+
+        return self._otel_context
+
+    @property
+    def baggage(self):
+        raise NotImplementedError
+        # TODO: Implement.
+
+
+class SpanWrapper(opentracing.Span):
+    def __init__(self, tracer, context, span):
+        self._otel_span = span
+        super().__init__(tracer, context)
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Span` object."""
+
+        return self._otel_span
+
+    def set_operation_name(self, operation_name):
+        self._otel_span.update_name(operation_name)
+
+        # Return self for call chaining.
+        return self
+
+    def finish(self, finish_time=None):
+        end_time = finish_time
+        if end_time is not None:
+            end_time = util.time_seconds_to_ns(finish_time)
+        self._otel_span.end(end_time=end_time)
+
+    def set_tag(self, key, value):
+        self._otel_span.set_attribute(key, value)
+
+        # Return self for call chaining.
+        return self
+
+    def log_kv(self, key_values, timestamp=None):
+        if timestamp is not None:
+            event_timestamp = util.time_seconds_to_ns(timestamp)
+        else:
+            event_timestamp = util.time_ns()
```

Comment:
I'd prefer passing `None` on to the OTel event if that is possible, as different API implementations might be be able to use a more precise/different timer then.

Reply:
Makes sense to me. At the moment the SDK will just leave `timestamp` as `None` though. Should I add a check at https://github.com/open-telemetry/opentelemetry-python/blob/b0da53d28b041338e04982f3a9715926887b5cad/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L211 similar to the following?

```python
if event.timestamp is None:
    event._timestamp = util.time_ns()
```

Or maybe add a timestamp setter to the `Event` class?

Reply:
Hmm, I don't know the spec on this but I'd add the check for None in `Event.__init__`. Seems like a bug to me if this is missing.

Reply:
@Oberon00 I've elaborated on the issue in https://github.com/open-telemetry/opentelemetry-python/issues/216 and proposed a solution. I will add my proposed solution to this PR and tag you for a re-review.

Reply:
Done @Oberon00.